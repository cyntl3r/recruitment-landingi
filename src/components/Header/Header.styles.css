.link {
  color: #000000;
  text-decoration: none;
}

.title {
  font-size: 1.1rem;
}

.title::before {
  content: 'Â» ';
}
