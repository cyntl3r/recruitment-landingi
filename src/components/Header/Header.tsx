import { FunctionComponent } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { HeaderState } from '../../reducers/headerReducer';
import { AppState } from '../../store';
import styles from './Header.styles.css';

export const Header: FunctionComponent = () => {
  const { title } = useSelector<AppState, HeaderState>((state) => state.header);
  return (
    <h1>
      <Link to="/" className={styles.link}>
        Blog
      </Link>{' '}
      {title && <span className={styles.title}>{title}</span>}
    </h1>
  );
};
