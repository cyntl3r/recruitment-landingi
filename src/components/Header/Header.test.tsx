import { render } from '../../testing/test-utils';
import { Header } from './Header';

test('snapshot', () => {
  const { container } = render(<Header />);
  expect(container).toMatchSnapshot();
});

test('wrapper render', () => {
  const { getByTestId } = render(<Header />);
  expect(getByTestId('header')).toBeDefined();
});
