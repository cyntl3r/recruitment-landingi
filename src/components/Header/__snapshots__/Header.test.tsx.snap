// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <h1
    data-testid="header"
  >
    <a
      href="/"
    >
      Blog
    </a>
     
    
  </h1>
</div>
`;
